### PR TITLE
Initial flow usage

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Add the configuration and the initial concept proof of typechecking with flow. It depends on mattermost/mattermost-redux#380

#### Ticket
[XYZ-65](https://mattermost.atlassian.net/browse/XYZ-65)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [ ] Ran `make flow` to ensure type checking passed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,17 +54,19 @@
     "xregexp": "3.2.0"
   },
   "devDependencies": {
-    "babel-cli": "6.26.0",
+    "babel-cli": "^6.26.0",
```
Comment (jwilander) [inline package.json]: Can we remove the `^` from all these? Don't want them auto-updating

Comment (jespino) [reply package.json]: Sure

**Comment by crspeller** [comment]
How much time does the flow build step add to the webapp build?

**Comment by jespino** [comment]
I going to measure it.

**Comment by jespino** [comment]
* Run 1 (in flow branch): `npm run build  314,34s user 18,36s system 176% cpu 3:08,95 total`
* Run 2 (in master branch): `npm run build  374,04s user 21,55s system 147% cpu 4:28,03 total`
* Run 3 (in flow branch): `npm run build  269,25s user 19,32s system 188% cpu 2:33,46 total`
* Run 4 (in master branch): `npm run build  263,80s user 18,69s system 189% cpu 2:28,96 total`

Looks like is hasn't any relevant impact on the webapp compilation.

**Comment by jespino** [comment]
I have another PR (mattermost/mattermost-redux#380) which have more "flow" code than the current webapp PR.

Run 1 (in master branch): `npm run build  14,45s user 0,20s system 123% cpu 11,905 total`
Run 2 (in flow branch): `npm run build  18,78s user 0,22s system 114% cpu 16,531 total`
Run 3 (in master branch): `npm run build  14,08s user 0,22s system 123% cpu 11,573 total`
Run 4 (in flow branch): `npm run build  15,35s user 0,27s system 122% cpu 12,786 total`

Looks like there is not high impact in the performance.

**Comment by jespino** [comment]
It doesn't works well with React.Fragment component, I have to investigate it.

**Comment by crspeller** [comment]
Reopen when ready.

**Pull request closed without merging**